gradebook.cpp

```diff
@@ -7,9 +7,61 @@ grade_Book::grade_Book(){}
 
 grade_Book::grade_Book(std::vector<std::string> assignment_name_list, std::vector<std::string> type, std::vector<int> score_obtained, std::vector<int> max_score){}
        
-double grade_Book::getIndividualGrade(std::string assignment_name){return 0.0;}
-double grade_Book::getCategoryGrade(std::string assignment_type){return 0.0;}
-double grade_Book::getCourseGrade(int option){return 0.0;}
+double grade_Book::getIndividualGrade(std::string assignment_name){
+    //creates an idex of the passed in assignment name
+    int idx;
+    idx = getAssignmentIndex(assignment_name);
+
+
+    //goes to the score_obtained vector to obtain and return the score of that assignment
+    //score as an percentage
+    //double score = ( (score_obtainted[idx] / max_score[idx]) * 100);
+    double score = score_obtainted[idx];
+
+    //print results
+    //std::cout << score << std::endl; //printed
+    return score;
+
+}
+
+double grade_Book::getCategoryGrade(std::string assignment_type){
+    double scoreTotal, maxTotal;
+    scoreTotal = 0;
+    maxTotal = 0;
+    //goes through the scores, if the index of the type equals the passed in assignment type->
+    //it will add that score of the assignment to the scoreTotal, and the max score of the assignment to maxTotal
+    for(int i = 0; i < score_obtainted.size(); i++){
+        if(type[i] == assignment_type){
+            scoreTotal += score_obtainted[i];
+            maxTotal += max_score[i];
+        }
+    }
+
+    double catGrade = scoreTotal / maxTotal;
+
+    //print results
+    //std::cout << catGrade << std::endl;
+    return catGrade;
+}
+
+//all grades within their respective categories and a course overall
+//only category totals and course overall
+//only the course overall
+double grade_Book::getCourseGrade(){
+    double courseScore, courseTotal;
+    courseScore = 0;
+    courseTotal = 0;
+
+    double courseGrade = 0;
+
+    for(int i = 0; i < score_obtainted.size(); i++){
+        courseScore += score_obtainted[i];
+        courseTotal += max_score[i];
+    }
+
+    courseGrade = ((courseScore / courseTotal) * 100);
+    return courseGrade;
+}

```

Comment:
The problem with this method is that the "int option" parameter was removed, that parameter is needed so that the 3 comments above the method can happen. The three comments are what the options are and what the method should be able to do depending on what option is picked by the user. A way this can be done is by using a switch statement depending on which option is chosen (i.e case 1: case 2: etc.), with this you can include a default in the switch statement that can cout "Not an option". Have case 1 be line 47, case 2 should be line 48, case 3 should be line 49.  The code written from lines 51-63 does not take account of any sort weighting needed for the method.